pilot/pkg/model/context.go

```diff
@@ -859,14 +859,20 @@ func (node *Proxy) SetServiceInstances(serviceDiscovery ServiceDiscovery) {
 	node.ServiceInstances = instances
 }
 
-// SetWorkloadLabels will set the node.Metadata.Labels only when it is nil.
+// SetWorkloadLabels will set the node.Metadata.Labels with priority:
+// 1. Proxy workload labels.
+// 2. node meta labels.
 func (node *Proxy) SetWorkloadLabels(env *Environment) {
-	// First get the workload labels from node meta
-	if len(node.Metadata.Labels) > 0 {
-		return
+	labels := env.GetProxyWorkloadLabels(node)
```

Comment:
what if they remove a label? Won't we still keep it?

Reply:
we should keep the previous proxy labels in the Proxy object and diff with the new

Reply:
good catch, i will think about